src/main/java/ogc/rs/apiserver/handlers/TilesMeteringHandler.java

```diff
@@ -0,0 +1,152 @@
+package ogc.rs.apiserver.handlers;
+
+import static ogc.rs.common.Constants.*;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.shareddata.LocalMap;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Arrays;
+import java.util.List;
+import ogc.rs.apiserver.util.AuthInfo;
+import ogc.rs.apiserver.util.MeteringInfo;
+import ogc.rs.catalogue.CatalogueService;
+import ogc.rs.metering.MeteringService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class TilesMeteringHandler implements Handler<Void> {
+
+  private static final Logger LOGGER = LogManager.getLogger(TilesMeteringHandler.class);
+
+  private final CatalogueService catalogueService;
+  private final MeteringService meteringService;
+  private final Vertx vertx;
+  Promise<Void> promise = Promise.promise();
+
+  /**
+   * Constructs a TilesMeteringHandler instance.
+   *
+   * @param vertx the Vert.x instance used for shared data and scheduling periodic tasks.
+   * @param config a JsonObject containing configuration for initializing CatalogueService. This
+   *     handler initializes: 1. A connection to the CatalogueService using the provided
+   *     configuration. 2. A metering service proxy for publishing data to RMQ.
+   *     <p>A periodic task is scheduled every 2 seconds, which: - Retrieves the shared data map
+   *     (LocalMap) storing metering information. - Iterates through each entry in the map: -
+   *     Removes the entry after retrieving its value. - Logs the removed entry and its value. -
+   *     Publishes the metering data (converted to JSON) to RMQ. - Logs success or failure for each
+   *     message published to RMQ.
+   */
+  public TilesMeteringHandler(Vertx vertx, JsonObject config) {
+    this.vertx = vertx;
+    this.catalogueService = new CatalogueService(vertx, config);
+    this.meteringService = MeteringService.createProxy(vertx, METERING_SERVICE_ADDRESS);
+
+    // Set up a periodic task to clean up the shared data map
+    vertx.setPeriodic(
+        2000,
+        id -> {
+          LocalMap<MeteringInfo, Integer> meteringMap = vertx.sharedData().getLocalMap("NAME");
+          meteringMap
+              .keySet()
+              .forEach(
+                  key -> {
+                    Integer value = meteringMap.remove(key);
+                    if (value != null) {
+                      LOGGER.info(this + " removed " + key.toJson() + " " + value);
+                      meteringService
+                          .insertMeteringValuesInRmq(key.toJson())
```

Comment:
Once the key is removed, update the size to `v`. Maybe update the `toJson` method to do this